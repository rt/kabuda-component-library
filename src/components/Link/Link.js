import React from 'react';
import PropTypes from 'prop-types';
import Events from '../../utils/events';
import Base from '../Base';
import { stores } from '../../libs/home';

const getStateStore = stores.stateStore.getStateStore;


/**
 * a link might not have enough description for a11y, allow aria-label attribute
*/
class Link extends Base {
    
    static contextTypes = {
        history: PropTypes.object
    };

    constructor(props) {
        super(props);
        
    }

    handleClick = (event) => {
        if (this.props.onClick) {
            this.props.onClick(event);
            //return control to caller
            return;
        }

        if (Events.isModifiedEvent(event) || !Events.isLeftClickEvent(event)) {
            return;
        }

        if (event.defaultPrevented === true) {
            return;
        }

        event.preventDefault();

        let to = this.props.to;
        if (!this.props.ignoreHistory) {
            to = getStateStore().getRouteHistoryManager().getHistoryRoute(to);
        }
        this.context.history.push(to);
    };

    render() {
        const { to, children, ignoreHistory, ...props } = this.props;
        return <a data-e2e={this.e2e()} href={to} {...props} onClick={this.handleClick}>{children}</a>;
    }
}

Link.propTypes = {
    to: PropTypes.string,
    children: PropTypes.node,
    onClick: PropTypes.func,
    ignoreHistory: PropTypes.bool,
};

export default Link;
