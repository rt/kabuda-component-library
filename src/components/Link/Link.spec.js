import React from 'react';
import Link from './Link';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer.create(<Link to="http://www.pebblefields.com">PebbleFields</Link>).toJSON();
    expect(tree).toMatchSnapshot();
});

it('allows child elments', () => {
    const tree = renderer.create(<Link to="http://www.pebblefields.com"><b><i>PebbleFields</i></b></Link>).toJSON();
    expect(tree).toMatchSnapshot();
});
