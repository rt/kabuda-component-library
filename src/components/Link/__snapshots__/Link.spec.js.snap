// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`allows child elments 1`] = `
<a
  data-e2e="link"
  href="http://www.pebblefields.com"
  onClick={[Function]}
>
  <b>
    <i>
      PebbleFields
    </i>
  </b>
</a>
`;

exports[`renders correctly 1`] = `
<a
  data-e2e="link"
  href="http://www.pebblefields.com"
  onClick={[Function]}
>
  PebbleFields
</a>
`;
